app/views/multifactor_auths/mfa_prompt.html.erb

```diff
@@ -17,15 +17,30 @@
       </div>
   <% end %>
 
-  <% if @user.totp_enabled? %>
-    <%= form_tag @form_mfa_url, method: :post do %>
-      <div class="text_field">
-        <%= label_tag :otp, t("multifactor_auths.otp_code"), class: "form__label" %>
-        <%= text_field_tag :otp, "", class: "form__input", autofocus: true, autocomplete: :off %>
-      </div>
-      <div class="form_bottom">
-        <%= submit_tag t("authenticate"), data: {disable_with: t("form_disable_with")}, class: "form__submit" %>
+  <% if @user.totp_enabled? || @user.webauthn_only_with_recovery? %>
+    <div class="mfa__option">
+      <% if @user.totp_enabled? %>
+        <h2 class="page__subheading--block"> <%= t("multifactor_auths.otp_code") %></h2>
+      <% elsif @user.webauthn_only_with_recovery? %>
+        <h2 class="page__subheading--block"> <%= t("multifactor_auths.recovery_code") %></h2>
+      <% end %>
+      <div class="t-body">
+        <p><%= t("multifactor_auths.recovery_code_html") %></p>
       </div>
-    <% end %>
+      <%= form_tag @form_mfa_url, method: :post do %>
+        <div class="text_field">
+          <% if @user.totp_enabled? %>
+            <%= label_tag :otp, t('multifactor_auths.otp_or_recovery'), class: 'form__label' %>
+            <%= text_field_tag :otp, '', class: 'form__input', autofocus: true, autocomplete: :off %>
+          <% elsif @user.webauthn_only_with_recovery? %>
```

Comment:
Hot take: could we render this even if there aren't any recovery codes? A user could have generated codes and used them all... I don't think we should be expose any details of if the user has any codes.

Reply:
Chatted about this with Jenny, and we decided to keep it the way it is - specifically to not show the recovery code prompt if the user doesn't have any. The justification for this, is that it takes up half of the page, and would do so while serving the user no functionality. So if a user doesn't have recovery codes, it would just be confusing to have half the page prompting them for it. The simplest design philosophy for users is not showing them stuff that is irrelevant to them, which is what we are doing.